crates/bevy_core_pipeline/src/upscaling/upscaling.wgsl

```diff
@@ -0,0 +1,25 @@
+struct VertexOutput {
+    [[builtin(position)]]
+    position: vec4<f32>;
+    [[location(0)]]
+    uv: vec2<f32>;
+};
+
+[[stage(vertex)]]
+fn vs_main([[builtin(vertex_index)]] in_vertex_index: u32) -> VertexOutput {
+    let uv = vec2<f32>(f32((in_vertex_index << 1u) & 2u), f32(in_vertex_index & 2u));
+    let position = vec4<f32>(uv * vec2<f32>(2.0, -2.0) + vec2<f32>(-1.0, 1.0), 0.0, 1.0);
```

Comment:
Is this approach preferred to a hard-coded fullscreen triangle? Seems like it requires some more operations perhaps even though both are practically insignificant performance-wise I guess.

Reply:
I have no idea. 

Is the alternative you had in mind
```rs
var vertices: array<vec3, 6> = [ ... ];`
vertices[in_vertex_index]
```
?

Reply:
I mean a single triangle that is larger than the visible normalised device coordinates in x,y and is specially crafted so that the UVs work out. I thought this would have insignificant performance impact but after looking it up, it seems it can definitely make a significant performance difference!
https://twitter.com/SebAaltonen/status/705831150019866624?s=20&t=wCGuq2EdqA2jYKx6euNkHw
https://michaldrobot.com/2014/04/01/gcn-execution-patterns-in-full-screen-passes/

In x,y, normalised device coordinates are lower-left -1,-1 to upper right 1,1. And UVs are interpolated across a triangle. So, you need 3 vertices, you need to cover the NDC x,y space, and you want vertices to be 0,0 bottom-left to 1,1 top-right. Note that the length of the sides of NDC x,y are 2. If you put vertices at (-1,-1), (3, -1), (-1, 3) they form a right-angled triangle that covers the screen. The UVs should be (0,0), (2,0), (0,2) so that they interpolate correctly. So for vertex indices 0,1,2, that can be expressed as (unless I made mistakes):
```rust
    out.clip_position = vec4<f32>(
        f32(in.vertex_index & 1),
        f32(in.vertex_index >> 1),
        0.0,
        0.5
    ) * 4.0 - 1.0;
    out.uv = vec2<f32>(
        f32(in.vertex_index & 1),
        f32(in.vertex_index >> 1),
    ) * 2.0;
```
And then draw 3 vertices. The `* 4.0 - 1.0` being outside the vec4 is I believe because of SIMD.

Reply:
Ah, commented further above. This will be a common vertex stage so I think we should add a `fullscreen.wgsl` or something, with just the vertex stage, and use it via an import.